wweval/R/combine_and_summarize_ww_data.R

```
@@ -0,0 +1,413 @@
+#' particular location and forecast date to produce a single row of output
+#' metadata for each combination. Locations without wastewater data are
+#' indicated as such and have NAs for the wastewater specific entries. This
+#' will be used to get an overall summary across locations and forecast dates
+#' of wastewater characteristics
+#'
+#' @param forecast_dates The vector of character strings of all the forecast
+#' dates
+#' @param locations The vector of character strings of all the locations
+#' @param eval_output_subdir The outer subdirectory of the nested file structure
+#'
+#' @return ww_metadata: a tibble with a row for each forecast date location and
+#' columns that provide summaries of the wastewater data
+#' @export
+combine_and_summarize_ww_data <- function(forecast_dates,
+                                          locations,
+                                          eval_output_subdir) {
+  if (length(forecast_dates) != length(locations)) {
+    cli::cli_abort(
+      message = "Vector of forecast dates and locations must be equal in length"
+    )
+  }
+
+  ww_metadata <- tibble()
+  flag_failed_output <- tibble()
+
+  for (i in seq_along(forecast_dates)) {
+    this_forecast_date <- forecast_dates[i]
+    this_location <- locations[i]
+
+    fp_ww <- get_filepath(
+      eval_output_subdir,
+      scenario = "status_quo",
+      forecast_date = this_forecast_date,
+      model_type = "ww",
+      location = this_location,
+      output_type = glue::glue("input_ww_data"),
+      file_extension = "tsv"
+    )
+    fp_hosp <- get_filepath(
+      eval_output_subdir,
+      scenario = "status_quo",
+      forecast_date = this_forecast_date,
+      model_type = "ww",
+      location = this_location,
+      output_type = glue::glue("input_hosp_data"),
+      file_extension = "tsv"
+    )
+
+    if (file.exists(fp_ww)) {
+      safe_data_load <- purrr::safely(load_data_and_summarize)
```

This will recreate the safe version of the function at every step of the for-loop. I would create it outside this function (i.e. at the scope of the `combine_and_summarize_ww_data.R` file, and just not export it if you don't think it should be user-exposed.

---

Removed the safe altogether per our discussion. `load_and_summarize_data()` now returns a single row dataframe with the NAs for ww metrics if the wastewater data is an empty csv. 

This means that if there is another error, the pipeline will fail loudly (which is what we want here, bc it means the issue is something else).